test/helpers/SignedHelpers.sol

```diff
@@ -0,0 +1,214 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.28;
+
+import {EuclidianMod} from "../utils/math/EuclidianMod.sol";
+
+contract SignedHelpers is EuclidianMod {
+    int256 constant intMax = type(int256).max;
+    int256 constant intMin = type(int256).min;
+
+    function sHandleZero(int256 num) public pure returns (int256) {
+        return num = num == int256(0) ? int256(1) : num;
+    }
+
+    function sEliminateMulOverflow(int256 x, int256 y) public pure returns (int256, int256) {
+        y = y == -1 ? int256(1) : y;
+
+        if (x == 1 || y == 1) {
+            return (2, 2);
+        }
+
+        int256 maxDivY = intMax / y;
+
+        if (x > 0 && y > 0 && x > maxDivY) {
+            return (maxDivY - 1, y);
+        }
+
+        if (x < 0 && y < 0 && x < maxDivY) {
+            return (maxDivY + 1, y);
+        }
+
+        int256 minDivY = intMin / y;
+
+        if (x > 0 && y < 0 && x > minDivY) {
+            return (minDivY - 1, y);
+        }
+
+        if (x < 0 && y > 0 && x < minDivY) {
+            return (minDivY + 1, y);
+        }
+
+        return (x, y);
+    }
+
+    function sEliminateDivOverflow(int256 x, int256 y, int256 denominator)
+        public
+        pure
+        returns (int256, int256, int256)
+    {
+        int256 product = x * y;
+
+        if (product == intMin && denominator == -1) {
+            return (x, y, 1);
+        }
+
+        return (x, y, denominator);
+    }
+
+    function sFindMulOverflow(int256 x, int256 y) public pure returns (int256, int256) {
+        if ((x == intMin && y == -1) || (x == -1 && y == intMin)) {
+            return (x, y);
+        }
+
+        y = y == -1 ? int256(1) : y;
+        x = x == -1 ? int256(1) : x;
+
+        if (x == 1 || y == 1) {
+            return (2, intMax);
+        }
+
+        int256 maxDivY = intMax / y;
+
+        if (x > 0 && y > 0 && x <= maxDivY) {
+            return (maxDivY + 1, y);
+        }
+
+        if (x < 0 && y < 0 && x >= maxDivY) {
+            return (maxDivY - 1, y);
+        }
+
+        int256 minDivY = intMin / y;
+
+        if (x > 0 && y < 0 && x <= minDivY) {
+            return (minDivY + 1, y);
+        }
+
+        if (x < 0 && y > 0 && x >= minDivY) {
+            return (minDivY - 1, y);
+        }
+
+        return (x, y);
+    }
+
+    function sFindDivisionWithoutRemainder(int256 x, int256 y, int256 denominator) public pure returns (int256) {
+        if (x * y % denominator != 0) {
+            return x / denominator * denominator;
```

Comment:
please make it clear 

(x / denominator) * denominator

or 

x / (denominator * denominator)